v2/oauth/introduction.mdx

```diff
@@ -0,0 +1,170 @@
+
+## Tokens
+
+Tokens are strings that represent the authorization issued to the [**Client**](#client).
+They are mainly used to access to protected resources, on behalf of the [**Resource Owner**](#resource-owner).
+At the same time, tokens can be used to have more information about who the owner is.
+
+
+### OAuth2 Access Token
+
+This is the main token that is used to provide temporary access to protected resources.
+The **OAuth2 Access Token** is meant to be read and validated only by the [**Resource Server**](#resource-server).
+
+:::info
+
+This token is different from the **SuperTokens Session Access Token**.
+The latter is used in the **OAuth 2.0** authentication flows when you set the `authMode: 'header'` option in the `Session` recipe.
+
+:::
+
+### OAuth2 Refresh Token
+
+A token that can be used to get a new [**OAuth2 Access Token**](#oauth2-access-token) when the current one has expired.
+Using the refresh token will not require the user to re-authenticate.
+
+:::info
+
+This token is different from the **SuperTokens Session Refresh Token**.
+The latter is used in the **OAuth 2.0** authentication flows when you set the `authMode: 'header'` option in the `Session` recipe.
+
+:::
+
+### ID Token
+
+This token provides identity information about the [**Resource Owner**](#resource-owner).
+Unlike [**OAuth2 Access Tokens**](/#oauth2-access-token), the **ID Tokens** are meant to be read only by the [**Client**](#client).
+
+## Scopes
+
+Scopes define the range of access that the [**Client**](#client) is requesting on behalf of the [**Resource Owner**](#resource-owner).
+They specify what portions of the **Resource Owner’s** data the **Client** can access and what actions it can perform.
+
+For example, when a user grants a web application permission to read their email, the application might request the `email.read` scope.
+
+In a general authentication flow scopes get used in the following way:
+1. When the **Client** gets created a series of scopes will be specified to the **Authorization Server**
+2. During the login attempt the **Resource Owner** will be asked to grant the requested scopes
+3. The **Autorization Server** will use the granted scopes to generate an **OAuth2 Access Token**.
+4. The **Resource Server** will check the scopes of the **OAuth2 Access Token** and will only allow the requested actions.
+
+
+
+## Authorization Flows
+
+The **OAuth 2.0** protocol defines several *flows* to accommodate different use cases.
+They are a set of steps an **OAuth Client** has to perform in order to obtain an access token.
+
+Our implementation supports the following flow types:
+
+### [Authorization Code Grant](https://oauth.net/2/grant-types/authorization-code/)
+
+<img src="/img/oauth/authorization-code-flow.png" alt="Authorization Code Grant"/>
+
+This flow is best suited for **web applications** where the [**Client**](/#client) can securely store a client secret (for example in the application's backend).
```

Comment:
But if we use with PKCE, then there is no need for a client secret. So this can be a bit misleading.

Reply:
resolved

Reply:
All security guidance on PKCE explicitly state that it is not intended to replace the client-secret, so that's something we should take care to not suggest.